Sources/TestingMacros/Support/DiagnosticMessage+Diagnosing.swift

```diff
@@ -17,30 +17,34 @@ public import SwiftSyntax
 public import SwiftSyntaxMacros
 #endif
 
-/// Diagnose issues with the traits in a parsed attribute.
-///
-/// - Parameters:
-///   - traitExprs: An array of trait expressions to examine.
-///   - attribute: The `@Test` or `@Suite` attribute.
-///   - context: The macro context in which the expression is being parsed.
-func diagnoseIssuesWithTraits(in traitExprs: [ExprSyntax], addedTo attribute: AttributeSyntax, in context: some MacroExpansionContext) {
-  for traitExpr in traitExprs {
-    // At this time, we are only looking for .tags() and .bug() traits in this
-    // function.
-    guard let functionCallExpr = traitExpr.as(FunctionCallExprSyntax.self),
-          let calledExpr = functionCallExpr.calledExpression.as(MemberAccessExprSyntax.self) else {
-      continue
-    }
-
-    // Check for .tags() traits.
-    switch calledExpr.tokens(viewMode: .fixedUp).map(\.textWithoutBackticks).joined() {
-    case ".tags", "Tag.List.tags", "Testing.Tag.List.tags":
-      _diagnoseIssuesWithTagsTrait(functionCallExpr, addedTo: attribute, in: context)
-    case ".bug", "Bug.bug", "Testing.Bug.bug":
-      _diagnoseIssuesWithBugTrait(functionCallExpr, addedTo: attribute, in: context)
-    default:
-      // This is not a trait we can parse.
-      break
+extension AttributeInfo {
+  /// Diagnose issues with the traits in a parsed attribute.
+  ///
+  /// - Parameters:
+  ///   - context: The macro context in which the expression is being parsed.
+  func diagnoseIssuesWithTraits(in context: some MacroExpansionContext) {
+    for traitExpr in traits {
+      if let functionCallExpr = traitExpr.as(FunctionCallExprSyntax.self),
+         let calledExpr = functionCallExpr.calledExpression.as(MemberAccessExprSyntax.self) {
+        // Check for .tags() traits.
+        switch calledExpr.tokens(viewMode: .fixedUp).map(\.textWithoutBackticks).joined() {
+        case ".tags", "Tag.List.tags", "Testing.Tag.List.tags":
+          _diagnoseIssuesWithTagsTrait(functionCallExpr, addedTo: self, in: context)
+        case ".bug", "Bug.bug", "Testing.Bug.bug":
+          _diagnoseIssuesWithBugTrait(functionCallExpr, addedTo: self, in: context)
+        default:
+          // This is not a trait we can parse.
+          break
+        }
+      } else if let memberAccessExpr = traitExpr.as(MemberAccessExprSyntax.self) {
+        switch memberAccessExpr.tokens(viewMode: .fixedUp).map(\.textWithoutBackticks).joined() {
+        case ".serialized", "SerializationTrait.serialized", "Testing.SerializationTrait.serialized":
```

Comment:
The type is now named `ParallelizationTrait`, as of #385

Reply:
D'oh